CMakeLists.txt

```diff
@@ -318,7 +318,15 @@ else()
     usFunctionCheckCompilerFlags(-Wno-mismatched-tags US_CXX_FLAGS)
   endif()
 
-  usFunctionCheckCompilerFlags("-fvisibility=hidden -fvisibility-inlines-hidden" _have_visibility)
+  # Issue #56: Workaround a clang compiler issue (https://llvm.org/bugs/show_bug.cgi?id=26749).
+  # The typeinfo symbols do not have reliable symbol visibility. Instead of exporting all
+  # symbols only export the typeinfo symbols by default. See
+  # https://developer.apple.com/library/mac/technotes/tn2185/_index.html#//apple_ref/doc/uid/DTS10004200-CH1-SUBSECTION2
+  if(US_COMPILER_CLANG OR US_COMPILER_APPLE_CLANG)
```

Comment:
good catch. ```-fvisibility-ms-compat``` is only meant to be used with clang on Mac OS X. ```-fvisibility-ms-compat``` must be used for both Apple's Clang (which is shipped with Xcode) and clang from [the LLVM website](http://llvm.org/releases/download.html).
I thought that ```US_COMPILER_CLANG OR US_COMPILER_APPLE_CLANG``` would cover both these cases, but did not consider clang on Linux.

Does ```US_COMPILER_CLANG``` represent clang only on Linux?
Is only using ```US_COMPILER_APPLE_CLANG``` sufficient?